x-pack/test/functional/services/machine_learning/custom_urls.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function MachineLearningCustomUrlsProvider({ getService }: FtrProviderContext) {
+  const testSubjects = getService('testSubjects');
+
+  return {
+    async assertCustomUrlLabelValue(expectedValue: string) {
+      const actualCustomUrlLabel = await testSubjects.getAttribute(
+        'mlJobCustomUrlLabelInput',
+        'value'
+      );
+      expect(actualCustomUrlLabel).to.eql(expectedValue);
+    },
+
+    async setCustomUrlLabel(customUrlsLabel: string) {
+      await testSubjects.setValue('mlJobCustomUrlLabelInput', customUrlsLabel, {
+        clearWithKeyboard: true,
+      });
+      await this.assertCustomUrlLabelValue(customUrlsLabel);
+    },
+
+    async assertCustomUrlItem(index: number) {
+      await testSubjects.existOrFail(`mlJobEditCustomUrlInput_${index}`);
+    },
+
+    async addCustomUrl() {
+      await testSubjects.click('mlJobAddCustomUrl');
```

Comment:
We should assert that the click has done something, i.e. that the modal/flyout is closed. I would suggest to pass the locator of some `customUrlContainerSubj` (the caller can then submit the data-test-subj for the modal or the flyout) and do a
```
await testSubjects.missingOrFail(customUrlContainerSubj, { timeout: 10 * 1000 });
```

Reply:
Added in https://github.com/elastic/kibana/pull/52269/commits/e2d4e95043071436594a0a9fabdc8e77303a2ee2